test/e2e/testdata/data-transformation-artifact-repository-ref.yaml

```diff
@@ -1,7 +1,7 @@
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
-  name: data-transformation
+  name: data-transformation-artifact-repository-ref
```

Comment:
Issue was that a similar test ran immediately previous to this test with the same name. When this test was launched, the E2E suite deleted the previous pods in the background, causing the launch of these pods to get deadlocked in some way I don't fully understand.

Changing the name of this workflow (and hence the name of its pods) was enough to fix the test. However...